### ISSUE TITLE
[Bug] Firestore DB needs additional API support to ingest certain data types / formats

### ISSUE THREAD

**Issue by somangshu** [appsmithorg/appsmith#7176]
## Description

When the user tries to use unix time stamp `moment(EventAddStartDatePicker.selectedDate).unix()` in the firestore API, the DB treats is as a map instead of timestamp. Instead when the user tries to use the firestore wrapper to do so, the function works properly `admin.firestore.Timestamp(moment(EventAddStartDatePicker.selectedDate).unix(), 0)`

The user wants support for the Firestore fields such as references, geocoordinates, etc.

![CleanShot_2021-09-06_at_16 04 282x](https://user-images.githubusercontent.com/11089579/132244613-6018a3bf-e5db-444c-9774-6ce4b58e09b6.png)
![CleanShot_2021-09-06_at_16 02 292x](https://user-images.githubusercontent.com/11089579/132244601-cbf02026-4180-4c22-967e-1f7b46eb0c8d.png)

### Important Details

- Version: [Cloud / Self-Hosted vx.x]
- OS: [e.g.MacOSX]
- Environment [production, release, deploy preview]

**Comment by tazmaniax** [comment]
Yes, something like this might be useful at least for the Timestamp field building a document reference would also be needed
```
{{
{
  "code": EventAddCodeInput.text,
  "title": EventAddTitleInput.text,
  "start": EventAddStartDatePicker.selectedDate ? new admin.firestore.Timestamp(moment(EventAddStartDatePicker.selectedDate).unix(), 0),
  "end": EventAddEndDatePicker.selectedDate ? new admin.firestore.Timestamp(moment(EventAddEndDatePicker.selectedDate).unix(), 0)
  "extPlatform": EventAddExtPlatformSelect.selectedOptionValue,
  "extReference": EventAddExtReferenceInput.text,
  "createdBy": appsmith.user.email
}
}}
```

**Comment by tazmaniax** [comment]
Somewhat related is Firebase is moving to a [more modular API, v9](https://firebase.google.com/docs/web/learn-more?authuser=0). I wonder if you need to wrap this API then maybe the new v9 API might make that task easier.

**Comment by somangshu** [comment]
Thanks for the additional info here @tazmaniax

**Comment by tazmaniax** [comment]
This is similar to https://github.com/appsmithorg/appsmith/issues/3741

**Comment by tazmaniax** [comment]
Hi, is there a rough timeline when this might be resolved? thx!

**Comment by tazmaniax** [comment]
Now it's [Hactoberfest](https://hacktoberfest.appsmith.com/), maybe I need to roll up my sleeves! 😄

**Comment by GreenFlux** [comment]
Another user is requesting support for Firebase timestamps:
https://discord.com/channels/725602949748752515/1221163808706400317/1221163808706400317

**Comment by Naveen-Goud** [comment]
Hello @tazmaniax , I have raised the PR for this task, Could you please review it.
